workers/generic-worker/main.go

```diff
@@ -551,6 +553,11 @@ func RunWorker() (exitCode ExitCode) {
 				log.Printf("No task claimed. Idle for %v%v.%v", idleTime, remainingIdleTimeText, remainingTaskCountText)
 			}
 		}
+
+		if graceful.TerminationRequested {
```

Comment:
I've just realised, this probably isn't safe. This is the main go routine, that is reading `graceful.TerminationRequested`, which could be happening in parallel to the same variable being updated by the worker runner protocol go routine. So I think we'll need some synchronisation primitives around the updates to the exported variables of the package, or to signal events via channels, if we want to be a little more idiomatic (although that comes at the cost of a little extra complexity).

Reply:
Good point!  I added a mutex.